aws/resource_aws_batch_job_queue.go

```diff
@@ -0,0 +1,206 @@
+		Create: resourceAwsBatchJobQueueCreate,
+		Read:   resourceAwsBatchJobQueueRead,
+		Update: resourceAwsBatchJobQueueUpdate,
+		Delete: resourceAwsBatchJobQueueDelete,
+
+		Schema: map[string]*schema.Schema{
+			"compute_environments": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 3,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"priority": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"state": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{batch.JQStateDisabled, batch.JQStateEnabled}, true),
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsBatchJobQueueCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+	input := batch.CreateJobQueueInput{
+		ComputeEnvironmentOrder: createComputeEnvironmentOrder(d),
+		JobQueueName:            aws.String(d.Get("name").(string)),
+		Priority:                aws.Int64(int64(d.Get("priority").(int))),
+		State:                   aws.String(d.Get("state").(string)),
+	}
+	name := d.Get("name").(string)
+	out, err := conn.CreateJobQueue(&input)
+	if err != nil {
+		return fmt.Errorf("%s %q", err, name)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{batch.JQStatusCreating, batch.JQStatusUpdating},
+		Target:     []string{batch.JQStatusValid},
+		Refresh:    batchJobQueueRefreshFunc(conn, name),
+		Timeout:    10 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("[WARN] Error waiting for JobQueue state to be \"VALID\": %s", err)
+	}
+
+	arn := *out.JobQueueArn
+	log.Printf("[DEBUG] JobQueue created: %s", arn)
+	d.SetId(arn)
+
+	return resourceAwsBatchJobQueueRead(d, meta)
+}
+
+func resourceAwsBatchJobQueueRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	jq, err := getJobQueue(conn, d.Get("name").(string))
+	if err != nil {
+		return err
+	}
+	if jq == nil {
+		return fmt.Errorf("[WARN] Error reading JobQueue: \"%s\"", err)
+	}
+	d.Set("arn", jq.JobQueueArn)
+	d.Set("compute_environments", jq.ComputeEnvironmentOrder)
+	d.Set("name", jq.JobQueueName)
+	d.Set("priority", jq.Priority)
+	d.Set("state", jq.State)
+	return nil
+}
+
+func resourceAwsBatchJobQueueUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	updateInput := &batch.UpdateJobQueueInput{
+		ComputeEnvironmentOrder: createComputeEnvironmentOrder(d),
+		JobQueue:                aws.String(d.Get("name").(string)),
+		Priority:                aws.Int64(int64(d.Get("priority").(int))),
+		State:                   aws.String(d.Get("state").(string)),
+	}
+	_, err := conn.UpdateJobQueue(updateInput)
```

Comment:
Shouldn't we also wait here for the job until it goes from `batch.JQStatusUpdating` to `batch.JQStatusValid`?